fdbserver/Ratekeeper.actor.cpp

```diff
@@ -436,7 +436,11 @@ Ratekeeper::Ratekeeper(UID id, Database db)
                 SERVER_KNOBS->SPRING_BYTES_TLOG_BATCH,
                 SERVER_KNOBS->MAX_TL_SS_VERSION_DIFFERENCE_BATCH,
                 SERVER_KNOBS->TARGET_DURABILITY_LAG_VERSIONS_BATCH) {
-	tagThrottler = std::make_unique<TagThrottler>(db, id);
+	if (SERVER_KNOBS->GLOBAL_TAG_THROTTLING) {
+		tagThrottler = std::make_unique<GlobalTagThrottler>(db, id);
+	} else {
+		tagThrottler = std::make_unique<TagThrottler>(db, id);
```

Comment:
I haven't really thought about this much, but is there any value in running both throttlers? If our quota based throttling fails to protect the cluster, will we just rely on global ratekeeper?

Reply:
There are cases where we need the global ratekeeper even if we're also using the tag throttler. For example, if the cluster is running out of disk space, the tag throttler has no means of throttling in response.

Reply:
For sure, I was mainly curious whether that was our only backstop. You could imagine that if the global tag throttler failed to protect the cluster, we could choose to focus the throttling on a smaller set of tags that was causing the largest impact (i.e. by using the existing tag throttler). Letting it fall back to the global ratekeeper could cause significant degradation or effective unavailability to everybody.